src/SFA.DAS.FindEmploymentSchemes.Web/Controllers/SchemesController.cs

```diff
@@ -1,25 +1,34 @@
+
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Logging;
+using SFA.DAS.FindEmploymentSchemes.Web.Infrastructure;
 using SFA.DAS.FindEmploymentSchemes.Web.Models;
 using SFA.DAS.FindEmploymentSchemes.Web.Services;
 using SFA.DAS.FindEmploymentSchemes.Web.ViewModels;
 
+
 namespace SFA.DAS.FindEmploymentSchemes.Web.Controllers
 {
     public class SchemesController : Controller
     {
         private readonly ILogger<SchemesController> _logger;
+        private readonly IWebHostEnvironment _environment;
         private readonly IFilterService _filterService;
 
-        public SchemesController(ILogger<SchemesController> logger, IFilterService filterService)
+        public SchemesController(ILogger<SchemesController> logger,
+                                 IWebHostEnvironment environment,
+                                 IFilterService filterService)
         {
             _logger = logger;
+            _environment = environment;
             _filterService = filterService;
         }
 
         [ResponseCache(Duration = 60 * 60, Location = ResponseCacheLocation.Any, NoStore = false)]
         public IActionResult Home()
         {
+            Sitemap.Generate(_environment, new System.Uri($"{Request.Scheme}://{Request.Host}"));
```

Comment:
I know the sitemap generation is protected by a static flag, but the generation doesn't really belong here.

Can we move it so that it's part of the start-up process?

Reply:
Moved to startup using config to determine base url